master/internal/proxy/proxy.go

```diff
@@ -74,19 +87,23 @@ func (p *Proxy) Receive(ctx *actor.Context) error {
 	return nil
 }
 
-func (p *Proxy) getTarget(serviceName string) (*url.URL, bool) {
-	p.lock.RLock()
-	defer p.lock.RUnlock()
-	target, ok := p.services[serviceName]
-	return target, ok
+func (p *Proxy) getTargetURL(serviceName string) (*url.URL, bool) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	service, ok := p.services[serviceName]
+	service.LastRequested = time.Now()
+	// Make a copy to avoid callers mutating the url outside of this locked
+	// method.
+	sURL := *service.URL
+	return &sURL, ok
 }
 
 // Service a normal (non-CONNECT) HTTP request through the /proxy/:service/* route.
-func (p *Proxy) newProxyHandler(serviceKey string) echo.HandlerFunc {
+func (p *Proxy) newProxyHandler(serviceID string) echo.HandlerFunc {
 	return func(c echo.Context) error {
 		// Look up the service name in the url path.
-		serviceName := c.Param(serviceKey)
-		target, ok := p.getTarget(serviceName)
+		serviceName := c.Param(serviceID)
+		serviceURL, ok := p.getTargetURL(serviceName)
```

Comment:
question: how often is `newProxyHandler` and `newConnectHandler` called? Asking this to understand how often `service.LastRequested` is going to be updated.

Reply:
`newProxyHandler` and `newConnectHandler` are called once but the function they return is called for every request that goes to `/proxy/:service`.